src/main/java/com/redhat/sast/api/service/OshSchedulerService.java

```diff
@@ -0,0 +1,573 @@
+     * 4. Process each scan individually in separate transactions
+     * 5. Update cursor to (highest processed scan ID + 1)
+     *    - Cursor only advances when scans are actually processed
+     *    - Cursor remains unchanged if no finished scans found (may be in progress)
+     *
+     * PHASE 2 - Retry processing (if retry enabled):
+     * 1. Fetch failed scans from retry queue
+     * 2. Process retry scans with backoff enforcement
+     * 3. Update retry attempt counters and cleanup successes
+     *
+     * Transaction boundary: NEVER - scheduler must not participate in transactions
+     * Concurrency: SKIP - prevents overlapping executions if one takes too long
+     */
+    @Scheduled(every = "${osh.poll.interval:5m}", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
+    @Transactional(Transactional.TxType.NEVER)
+    public void pollOshForNewScans() {
+        if (!oshConfiguration.isEnabled()) {
+            LOGGER.trace("OSH integration disabled, skipping poll cycle");
+            return;
+        }
+
+        try {
+            LOGGER.debug("Starting OSH two-phase polling cycle");
+
+            ProcessingResults incrementalResults = processIncrementalScans();
+
+            ProcessingResults retryResults = processRetryScans();
+
+            logCombinedResults(incrementalResults, retryResults);
+
+        } catch (Exception e) {
+            LOGGER.error("OSH polling cycle failed, will retry next scheduled interval", e);
+        }
+    }
+
+    /**
+     * PHASE 1: Process incremental scans.
+     * Handles new scans from OSH with cursor advancement.
+     *
+     * Cursor advancement strategy:
+     * - Only advances when scans are successfully processed
+     * - Advances to (highest processed scan ID + 1)
+     * - Does NOT advance if no finished scans found (they might be in progress)
+     *
+     * @return processing results for incremental scans
+     */
+    private ProcessingResults processIncrementalScans() {
+        try {
+            LOGGER.debug("Starting incremental scan processing");
+
+            PollConfiguration config = preparePollConfiguration();
+            List<OshScanResponse> finishedScans = fetchFinishedScans(config);
+
+            if (finishedScans.isEmpty()) {
+                LOGGER.debug(
+                        "No finished scans found in range {}-{}, cursor unchanged (scans may be in progress)",
+                        config.startScanId(),
+                        config.startScanId() + config.batchSize() - 1);
+                return new ProcessingResults(0, 0, 0, PHASE_INCREMENTAL);
+            }
+
+            LOGGER.debug(
+                    "Found {} finished scans in range {}-{}",
+                    finishedScans.size(),
+                    config.startScanId(),
+                    config.startScanId() + config.batchSize() - 1);
+
+            ProcessingResults results = processScans(finishedScans, PHASE_INCREMENTAL);
```

Comment:
Why we define `PHASE_INCREMENTAL` as a parameter? We do not have other strategies.